src/CommunityToolkit.Maui.MediaElement/MediaElement.shared.cs

```diff
@@ -255,7 +255,11 @@ public bool ShouldShowPlaybackControls
 	public MediaSource? Source
 	{
 		get => (MediaSource)GetValue(SourceProperty);
-		set => SetValue(SourceProperty, value);
+		set
+		{
+			ClearTimer();
+			SetValue(SourceProperty, value);
```

Comment:
Just curious - is there a reason we're calling `ClearTimer()` here in the Property's setter instead of calling `ClearTimer()` in `void OnSourcePropertyChanged(MediaSource?)`?

Reply:
I'd prefer to put all of the logic `void OnSourcePropertyChanged(MediaSource?)` for consistency + easier maintainability if it still solves this bug.

Reply:
There was no specific reason for using the property setter instead of `void OnSourcePropertyChanged(MediaSource?)`.

I've moved the `ClearTimer()` to `void OnSourcePropertyChanged(MediaSource?)`... still solves the issues for me.

Reply:
Right on! 🙌